### PR TITLE
Update retrieval and datatransfer

### PR DESCRIPTION
**Pull request by turuslan**
### Description of the Change
Update retrieval.
Update datatransfer.

### CODE REVIEW

File: core/codec/cbor/cbor_decode_stream.hpp [hunk 1]
```diff
@@ -141,6 +141,8 @@ namespace fc::codec::cbor {
     std::vector<uint8_t> raw();
     /** Creates map container decode substream map */
     std::map<std::string, CborDecodeStream> map();
+    static CborDecodeStream &named(std::map<std::string, CborDecodeStream> &map,
```
Comment (artyom-yurin) [inline core/codec/cbor/cbor_decode_stream.hpp:144]: Maybe better name GetByName or something like this

File: core/codec/cbor/cbor_errors.cpp [hunk 2]
```diff
@@ -32,6 +32,8 @@ OUTCOME_CPP_DEFINE_CATEGORY(fc::codec::cbor, CborDecodeError, e) {
       return "Invalid CID";
     case CborDecodeError::kWrongSize:
       return "Wrong size";
+    case CborDecodeError::kKey:
```
Comment (artyom-yurin) [inline core/codec/cbor/cbor_errors.cpp:35]:
```suggestion
    case CborDecodeError::kNotFound :
```

**Review by artyom-yurin — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]
Mostly naming remarks.

File: core/markets/storage/chain_events/impl/chain_events_impl.cpp [hunk 3]
```diff
@@ -78,7 +79,7 @@ namespace fc::markets::storage::chain_events {
   };
 
   outcome::result<void> ChainEventsImpl::onMessage(
-      const UnsignedMessage &message) {
+      const UnsignedMessage &message, const CID &cid) {
```
[the remark this answers is not in the document]
Comment (turuslan) [reply core/markets/storage/chain_events/impl/chain_events_impl.cpp:82]: `onMessage` can't accepts both `SignedMessage` and `UnsignedMessage`, so cid is passed separately

**Pull request merged**